### PR TITLE
`getTransactionByHash` don't find tx in mempool

### LINKED ISSUE #260: `getTransactionByHash` don't find tx in mempool
__Proposal:__ [Description of the feature]

__Current behavior:__ [What currently happens]

__Desired behavior:__ [What you would like to happen]

__Use case:__ [Why is this important (helps with prioritizing requests)]

which is the case with [go-ethereum](https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1539)

### PR DESCRIPTION
**Pull request by yihuang**
Closes #260

## Description

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: ethereum/rpc/namespaces/eth/api.go [hunk 1]
```diff
@@ -604,7 +604,47 @@ func (e *PublicAPI) GetTransactionByHash(hash common.Hash) (*rpctypes.RPCTransac
...
+				from,
+				msg.AsTransaction().Hash(),
+				common.Hash{},
+				uint64(0),
+				uint64(0),
+			)
+			if err != nil {
+				return nil, err
+			}
+			return rpctx, nil
```
Comment (fedekunze) [inline ethereum/rpc/namespaces/eth/api.go:646]: this is not filtering the tx by the hash

Comment (yihuang) [reply ethereum/rpc/namespaces/eth/api.go]: good catch, fixed 😄

File: ethereum/rpc/namespaces/eth/api.go [hunk 2]
```diff
@@ -604,7 +604,54 @@ func (e *PublicAPI) GetTransactionByHash(hash common.Hash) (*rpctypes.RPCTransac
...
+				hash,
+				common.Hash{},
+				uint64(0),
+				uint64(0),
+			)
+			if err != nil {
+				return nil, err
+			}
+			return rpctx, nil
+		}
```
Comment (fedekunze) [inline ethereum/rpc/namespaces/eth/api.go:654]: this part is shared with the one below, do you think we can refactor it to reduce the code?

Comment (yihuang) [reply ethereum/rpc/namespaces/eth/api.go:654]: yes, you can merge one first, then I refactor the other one.

**Review by fedekunze — APPROVED** [review]

**Pull request merged**